examples/powershell/vro/handler/script.ps1

```diff
@@ -0,0 +1,82 @@
+
+# Process function Secrets passed in
+$SECRETS_FILE = "/var/openfaas/secrets/vro-secrets"
+$SECRETS_CONFIG = (Get-Content -Raw -Path $SECRETS_FILE | ConvertFrom-Json)
+
+# Process payload sent from vCenter Server Event
+$json = $args | ConvertFrom-Json
+if($env:function_debug -eq "true") {
+    Write-Host "DEBUG: json=`"$($json | Format-List | Out-String)`""
+}
+
+$vcenter = ($json.source -replace "https://","" -replace "/sdk","");
+$vmMoRef = $json.data.vm.vm.value;
+$vm = $json.data.vm.name;
```

Comment:
There's no safety checks here validating that these properties exist. Would it be better to throw an informative error (e.g. if someone triggers on the wrong event type) instead of blowing up the function (exception)? Or does PS handle missing (NULL) fields correctly and stop processing?

Reply:
👍 Just did a check if either are NULL, then print and exit

Reply:
Final question: PS json module treats NULL and empty string "" as equivalent? I.e. if the string is not present it will be marshalled to `""` ?

Reply:
That's correct, rather than throw an error unlike other languages. Good/Bad :)